addons/mail/static/src/core/public_web/thread_actions_patch.js

```diff
@@ -0,0 +1,12 @@
+import { threadActionsRegistry } from "@mail/core/common/thread_actions";
+import { _t } from "@web/core/l10n/translation";
+
+threadActionsRegistry.add("leave", {
+    condition: (component) =>
+        component.ui.isSmall && (component.thread?.canLeave || component.thread?.canUnpin),
+    icon: "fa fa-fw fa-sign-out text-danger",
+    name: (component) => (component.thread.canLeave ? _t("Leave") : _t("Unpin")),
+    open: (component) =>
+        component.thread.canLeave ? component.thread.leaveChannel() : component.thread.unpin(),
+    sequence: 100,
```

Comment:
Would put:
```js
 sequence: 10,
 sequenceGroup: 40,
```
So that it's in the 4th group. extra 0 in numbered sequence so that new users can easily put sequence before/after/in-between.

FYI the grouping is loosely chosen as follow:
- sequenceGroup 10: panels to view extra content: threads, pinned messages, attachments, members
- sequenceGroup 20: actions that affect conversation in a relatively big way and / or change screen navigation enough: invite people, open in discuss / form view, search messages
- sequenceGroup 30: the settings
- sequenceGroup 40: high commitment destructive actions, like the leave